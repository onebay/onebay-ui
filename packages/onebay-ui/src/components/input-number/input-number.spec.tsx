import { mount } from '@vue/test-utils'
import InputNumber from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(InputNumber, {
    slots,
    props: { ...values },
  })
}

describe('InputNumber Snap', () => {
  it('render initial InputNumber', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props type(number)', () => {
    const wrapper = factory({ type: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props type(digit)', () => {
    const wrapper = factory({ type: 'digit' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props disabledInput', () => {
    const wrapper = factory({ disabledInput: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props value', () => {
    const wrapper = factory({ value: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props width', () => {
    const wrapper = factory({ width: 200 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props size', () => {
    const wrapper = factory({ size: 'large' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props step', () => {
    const wrapper = factory({ step: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props value & min = ""', () => {
    const wrapper = factory({ value: '', min: 0 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render InputNumber -- props value = "a"', () => {
    const wrapper = factory({ value: 'a' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('InputNumber Event', () => {
  it('InputNumber onChange', async () => {
    const wrapper = factory({ value: 2, disabled: true })
    wrapper.find('.ob-input-number .ob-input-number__input').trigger('input')
    expect(wrapper.emitted()).not.toHaveProperty('change')
    await wrapper.setProps({ disabled: false })
    wrapper.find('.ob-input-number .ob-input-number__input').trigger('input')
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('InputNumber onBlur', () => {
    const onBlur = jest.fn()
    const wrapper = factory({ value: 2, onBlur: onBlur })
    wrapper.find('.ob-input-number .ob-input-number__input').trigger('blur')
    expect(onBlur).toBeCalled()
  })

  it('InputNumber onClick min bigger than value', async () => {
    const wrapper = factory({ value: 2, min: 3 })
    wrapper.find('.ob-input-number .ob-input-number__btn').trigger('click')
    expect(wrapper.emitted()).not.toHaveProperty('change')
    await wrapper.setProps({ min: 0 })
    wrapper.find('.ob-input-number .ob-input-number__btn').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('InputNumber onClick max smaller than value', async () => {
    const wrapper = factory({ value: '', max: 18, disabled: true })
    wrapper.find('.ob-input-number .ob-input-number__btn:last-child').trigger('click')
    expect(wrapper.emitted()).not.toHaveProperty('change')
    await wrapper.setProps({ disabled: false, value: 20 })
    wrapper.find('.ob-input-number .ob-input-number__btn:last-child').trigger('click')
    expect(wrapper.emitted()).not.toHaveProperty('change')
    await wrapper.setProps({ value: 10 })
    wrapper.find('.ob-input-number .ob-input-number__btn:last-child').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('change')
  })
})
