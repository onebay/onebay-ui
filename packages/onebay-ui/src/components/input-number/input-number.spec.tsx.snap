// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber Snap render InputNumber -- props className 1`] = `
<div
  class="ob-input-number test"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props customStyle 1`] = `
<div
  class="ob-input-number"
  style="color: red;"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props disabled 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn ob-input-number--disabled"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    disabled=""
    type="number"
  />
  <div
    class="ob-input-number__btn ob-input-number--disabled"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props disabledInput 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    disabled=""
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props size 1`] = `
<div
  class="ob-input-number ob-input-number--lg"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props step 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props type(digit) 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="digit"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props type(number) 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props value & min = "" 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn ob-input-number--disabled"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props value = "a" 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn ob-input-number--disabled"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props value 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render InputNumber -- props width 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    style="width: 200px;"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;

exports[`InputNumber Snap render initial InputNumber 1`] = `
<div
  class="ob-input-number"
>
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-subtract ob-input-number__btn-subtract"
    />
  </div>
  <input
    class="ob-input-number__input"
    type="number"
  />
  <div
    class="ob-input-number__btn"
  >
    <div
      class="ob-icon ob-icon-add ob-input-number__btn-add"
    />
  </div>
</div>
`;
